org.jacoco.core.test/src/org/jacoco/core/internal/BytecodeVersionTest.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal;
+
+import org.junit.Test;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+
+public class BytecodeVersionTest {
```

Comment:
Add explicit tests for read/write methods?

Reply:
@marchof while adding test, I realized that this is only about major segment of version, i.e.

```
BytecodeVersion.get(createClass(Opcodes.V1_1)) != Opcodes.V1_1
```

this is consistent with your request https://github.com/jacoco/jacoco/pull/629#discussion_r175668010 to use `BytecodeVersion.get` in `ProbeArrayStrategyFactory` because reading of both major and minor will break comparisons in it - counterintuitively `Opcodes.V1_1` is greater than `Opcodes.V9` and `BytecodeVersion.V10`.

I think that for the time being will be enough just to document this and reflect in test. WDYT?

Reply:
@Godin Just add to JavaDoc that get/set is for major version only. The day when OpenJDK releases weekly with minor versions we think about it again ;)

Reply:
@marchof test added, JavaDoc updated